### PR TITLE
Gradle updates

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Comment by sschuberth** [comment]
Looks like this will conflict with #253 once merged.

File: build.gradle [hunk 1]
```diff
@@ -8,7 +8,7 @@ buildscript {
     dependencies {
         classpath 'com.android.tools.build:gradle:2.3.3'
         classpath 'com.dicedmelon.gradle:jacoco-android:0.1.1'
-
+        classpath 'com.github.ben-manes:gradle-versions-plugin:0.15.0'
```
Comment (yoavst) [inline build.gradle]: Explanation?

**Comment by yoavst** [comment]
I can merge your changes to my pr.  @kittinunf do we want the versions on `gradle.properties` or in main `build.gradle`?

File: build.gradle (hunk 1 above)
Comment (sschuberth) [reply build.gradle]: Please see the commit message for that commit.

**Comment by kittinunf** [comment]
In my project, I don't put the version of dependencies in `gradle.properties` but I might have overlooked something.

I would love to hear whether there is a big win of doing this?

**Comment by sschuberth** [comment]
IMO, putting project-wide (static) properties in `gradle.properties` is cleaner. Like the file name says, this simply is where they belong to.

**Comment by kittinunf** [comment]
Let's revisit this later as it looks stale.

**Comment by sschuberth** [comment]
No idea why you believe this to look stale from my side. I was replying to your question, and you did not respond any further to my argument.

**Pull request closed without merging**